runtime/queries/metricsview_toplist.go

```diff
@@ -67,229 +65,167 @@ func (q *MetricsViewToplist) UnmarshalResult(v any) error {
-	olap, release, err := rt.OLAP(ctx, instanceID, q.MetricsView.Connector)
+	ds := []*runtimev1.MetricsViewAggregationDimension{{Name: q.DimensionName}}
+	ms := make([]*runtimev1.MetricsViewAggregationMeasure, len(q.MeasureNames))
+	for i, m := range q.MeasureNames {
+		ms[i] = &runtimev1.MetricsViewAggregationMeasure{Name: m}
+	}
+
+	mv, security, err := resolveMVAndSecurityFromAttributes(ctx, rt, instanceID, q.MetricsViewName, q.SecurityAttributes, q.SecurityPolicy, ds, ms)
 	if err != nil {
 		return err
 	}
-	defer release()
-
-	switch olap.Dialect() {
-	case drivers.DialectDuckDB:
-		if opts.Format == runtimev1.ExportFormat_EXPORT_FORMAT_CSV || opts.Format == runtimev1.ExportFormat_EXPORT_FORMAT_PARQUET {
-			if q.MetricsView.TimeDimension == "" && (q.TimeStart != nil || q.TimeEnd != nil) {
-				return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsViewName)
-			}
-
-			sql, args, err := q.buildMetricsTopListSQL(q.MetricsView, olap.Dialect(), q.ResolvedMVSecurity)
-			if err != nil {
-				return err
-			}
-
-			filename := q.generateFilename()
-			if err := DuckDBCopyExport(ctx, w, opts, sql, args, filename, olap, opts.Format); err != nil {
-				return err
-			}
-		} else {
-			if err := q.generalExport(ctx, rt, instanceID, w, opts); err != nil {
-				return err
-			}
-		}
-	case drivers.DialectDruid, drivers.DialectClickHouse, drivers.DialectPinot:
-		if err := q.generalExport(ctx, rt, instanceID, w, opts); err != nil {
-			return err
-		}
-	default:
-		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
-	}
-
-	return nil
-}
 
-func (q *MetricsViewToplist) generalExport(ctx context.Context, rt *runtime.Runtime, instanceID string, w io.Writer, opts *runtime.ExportOptions) error {
-	err := q.Resolve(ctx, rt, instanceID, opts.Priority)
+	qry, err := q.rewriteToMetricsViewQuery(true)
 	if err != nil {
-		return err
+		return fmt.Errorf("error rewriting to metrics query: %w", err)
 	}
 
-	if opts.PreWriteHook != nil {
-		err = opts.PreWriteHook(q.generateFilename())
-		if err != nil {
-			return err
-		}
+	e, err := metricsview.NewExecutor(ctx, rt, instanceID, mv, security, opts.Priority)
+	if err != nil {
+		return err
 	}
+	defer e.Close()
 
+	var format string
 	switch opts.Format {
-	case runtimev1.ExportFormat_EXPORT_FORMAT_UNSPECIFIED:
-		return fmt.Errorf("unspecified format")
 	case runtimev1.ExportFormat_EXPORT_FORMAT_CSV:
-		return WriteCSV(q.Result.Meta, q.Result.Data, w)
+		format = "csv"
 	case runtimev1.ExportFormat_EXPORT_FORMAT_XLSX:
-		return WriteXLSX(q.Result.Meta, q.Result.Data, w)
+		format = "xlsx"
 	case runtimev1.ExportFormat_EXPORT_FORMAT_PARQUET:
-		return WriteParquet(q.Result.Meta, q.Result.Data, w)
+		format = "parquet"
+	default:
+		return fmt.Errorf("unsupported format: %s", opts.Format.String())
 	}
 
-	return nil
-}
+	path, err := e.Export(ctx, qry, nil, format)
```

Comment:
Why not  directly use `runtimev1.ExportFormat` as `parameter` type ?

Reply:
It's an API message so would be slightly weird to pass into a model's `OutputProperties`. Just added a designated `drivers.FileFormat` type to use instead.